src/sentry/static/sentry/app/views/performance/transactionSummary/vitalsChart.tsx

```diff
@@ -0,0 +1,254 @@
+import React from 'react';
+import {browserHistory} from 'react-router';
+import * as ReactRouter from 'react-router';
+import {Location} from 'history';
+
+import {OrganizationSummary} from 'app/types';
+import {Client} from 'app/api';
+import {t} from 'app/locale';
+import LineChart from 'app/components/charts/lineChart';
+import ChartZoom from 'app/components/charts/chartZoom';
+import ErrorPanel from 'app/components/charts/errorPanel';
+import TransparentLoadingMask from 'app/components/charts/transparentLoadingMask';
+import TransitionChart from 'app/components/charts/transitionChart';
+import EventsRequest from 'app/components/charts/eventsRequest';
+import ReleaseSeries from 'app/components/charts/releaseSeries';
+import QuestionTooltip from 'app/components/questionTooltip';
+import {getInterval} from 'app/components/charts/utils';
+import {IconWarning} from 'app/icons';
+import {getUtcToLocalDateObject} from 'app/utils/dates';
+import EventView from 'app/utils/discover/eventView';
+import {getMeasurementSlug, getAggregateArg} from 'app/utils/discover/fields';
+import withApi from 'app/utils/withApi';
+import {decodeScalar} from 'app/utils/queryString';
+import theme from 'app/utils/theme';
+import {tooltipFormatter, axisLabelFormatter} from 'app/utils/discover/charts';
+import getDynamicText from 'app/utils/getDynamicText';
+
+import {HeaderTitleLegend} from '../styles';
+
+const QUERY_KEYS = [
+  'environment',
+  'project',
+  'query',
+  'start',
+  'end',
+  'statsPeriod',
+] as const;
+
+type ViewProps = Pick<EventView, typeof QUERY_KEYS[number]>;
+
+type Props = ReactRouter.WithRouterProps &
+  ViewProps & {
+    api: Client;
+    location: Location;
+    organization: OrganizationSummary;
+  };
+
+const YAXIS_VALUES = [
+  'p75(measurements.fp)',
+  'p75(measurements.fcp)',
+  'p75(measurements.lcp)',
+  'p75(measurements.fid)',
+];
+
+class VitalsChart extends React.Component<Props> {
+  handleLegendSelectChanged = legendChange => {
+    const {location} = this.props;
+    const {selected} = legendChange;
+    const unselected = Object.keys(selected).filter(key => !selected[key]);
+
+    const to = {
+      ...location,
+      query: {
+        ...location.query,
+        unselectedSeries: unselected,
+      },
+    };
+    browserHistory.push(to);
+  };
+
+  render() {
+    const {
+      api,
+      project,
+      environment,
+      location,
+      organization,
+      query,
+      statsPeriod,
+      router,
+    } = this.props;
+
+    const unselectedSeries = location.query.unselectedSeries ?? [];
+    const unselectedMetrics = Array.isArray(unselectedSeries)
+      ? unselectedSeries
+      : [unselectedSeries];
+    const seriesSelection = unselectedMetrics.reduce((selection, metric) => {
+      selection[metric] = false;
+      return selection;
+    }, {});
```

Comment:
We have similar blocks in a few charts, should we consider making a function that takes the location and returns the seriesSelection? Maybe something to do separately.